### ISSUE TITLE
TensorFlow Lite object detection example code does not work with custom trained model

### ISSUE THREAD

**Issue by emre84** [tensorflow/tensorflow#47779]
### System information

-   **OS Platform and Distribution (macOS Big Sur 11.2)**:
-   **Mobile device (iPhone 7)
-   **TensorFlow Lite installed from (https://www.tensorflow.org/lite/examples/object_detection/overview:View iOS example)**:
-   **TensorFlow (1.14.1)
-   **Xcode version(Version 12.4)
-   **Python (Python 3.7.3)

### Describe the problem
I want to test my custom trained (ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync) model with tensorflow lite, to see if the performance (accuracy and fps) is good enough to be able to start the development of my project for IOS devices or not. But when I execute my .tflite and labelmap.txt files with TensorFlow Lite example code, nothing can be detected.

Here is my flow:

1- I trained my images with ssd_mobilenet_

v1_fpn_shared_box_predictor_640x640_coco14_sync model:
Code:
https://github.com/tensorflow/models
(master branch)
Command:
python3.7 train.py
--logtostderr
--train_dir=/Users/Documents/Temp/tensorflow_last/models/train
--pipeline_config_path=/Users/Documents/Temp/tensorflow_last/models/ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync.config
Note:I can also attach my config file, if it is related

2-Converted it to .pb file (tflite_graph.pb):
python3.7 models/research/object_detection/object_detection/export_tflite_ssd_graph.py
--pipeline_config_path=/Users/emre/Documents/Temp/tensorflow_last/models/ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync.config
--trained_checkpoint_prefix=/Users/emre/Documents/Temp/tensorflow_last/models/train/model.ckpt-21881
--output_directory=/Users/emre/Documents/Temp/tensorflow_last/models/Lite

3-Converted it to tflite file:
Code:
https://github.com/tensorflow/models
(master branch)
Command:
python3.7 tflite_convert.py --output_file=test.tflite --graph_def_file=/Users/Documents/Temp/tensorflow_last/models/Lite/tflite_graph.pb --input_arrays=image_tensor --output_arrays='detection_boxes','detection_scores','detection_classes','num_detections' --input_shape=1,800,600,3 --allow_custom_ops

4-Downloaded following Tensorflow example code from https://www.tensorflow.org/lite/examples/object_detection/overview
(View iOS example)

The example works fine with the existing pretrained model(which is in tensorflow_lite/examples/lite/examples/object_detection/ios/ObjectDetection/Model/detect.tflite and labelmap.txt) on Xcode version(Version 12.4) and on Mobile device (iPhone 7)).

5-Changed original detect.tflite and labelmap.txt files with my own test.tflite(re-named it as detect.tflite) and labelmap.txt. But when I try to execute the same code to detect my own images with my own .tflite and label map.txt files, nothing can be detected and no error messages pop up.

6-Also tried to execute following pre-trained models with Tensorflow example code, to see if there is a problem with my own custom trained model or not:
https://github.com/asus4/tf-lite-unity-sample/tree/master/Assets/StreamingAssets/mediapipe
But they did not also work properly. Nothing can be detected and no error messages pop up.

I assume that I miss something to modify in example code, since pre-trained models from github.com can also not detect any objects.

Can someone please help me if I need anything else to modify to make my model work? 

Thanks a lot in advance.

### Source code / logs
-No Related log file can I observe in Xcode(Nothing special in debugger output or device logs) 
Note: I have no experience with Xcode. I hope I do not miss any points. I can provide any logs, if you require.

**Comment by abattery** [comment]
@lintian06 could you take a look at this?

**Comment by lintian06** [comment]
Hi @emre84 ,

As the model is changed, please notice that it has different inputs and outputs from our built-in one. Therefore, most likely, you need to change those inputs and outputs in [ModelDataHandler](https://github.com/tensorflow/examples/blob/f2946cb116ac5e06eb3e318dd94144214c95238a/lite/examples/object_detection/ios/ObjectDetection/ModelDataHandler/ModelDataHandler.swift#L46-L63) accordingly.

Seeing from your conversion script, you may need to change the swift code to adapt to your new model.
```
--input_shape=1,800,600,3 
# Change your input shape to 800x600 in the swift code.

--output_arrays='detection_boxes','detection_scores','detection_classes','num_detections'
# Please make sure the meanings of output values and orders are aligned with swift code, or otherwise adapt to it. The model may be varied a bit from case to case, which I cannot tell by the name.

--allow_custom_ops
# I would suggest removing this to see if the conversion still works for your model. FYI, if [custom ops](https://www.tensorflow.org/lite/guide/ops_custom) have to be there, it requires a bit more efforts.
```

Please use xcode's console to debug, the app doesn't show errors in the GUI.
